TTHAnalysis/python/plotter/makeShapeCards.py

```diff
@@ -124,19 +126,19 @@ def rebin2Dto1D(h,funcstring):
     newh.SetLineColor(h.GetLineColor())
     return newh
 
-myout = outdir+"/common/" if len(masses) > 1 else outdir;
+myout = os.path.join(options.outdir,"common") if len(masses) > 1 else options.outdir;
 
 report={}
 if options.infile!=None:
-    infile = ROOT.TFile(myout+binname+".bare.root","read")
+    infile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"read")
     for p in mca.listSignals(True)+mca.listBackgrounds(True)+['data']:
         h = infile.Get(p)
         if h: report[p] = h
 else:
     report = mca.getPlotsRaw("x", args[2], args[3], cuts.allCuts(), nodata=options.asimov)
 
 if options.savefile!=None:
-    savefile = ROOT.TFile(myout+binname+".bare.root","recreate")
+    savefile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"recreate")
     for n,h in report.iteritems(): savefile.WriteTObject(h,n)
```

Comment:
The code was doing what it was supposed to, since we had been using it before.
You have to learn to avoid polluting the git history when modifying files that are not used just by you, and it looks like if I let PRs pass through you don't learn.
Even the commit history of this PR is a mess, with lots of duplicated commits.